Content.Client/CombatMode/ShowCombatModeIndicatorsOverlay.cs

```diff
@@ -0,0 +1,122 @@
+    private CombatModeSystem _combatSystem;
+
+    public override OverlaySpace Space => OverlaySpace.ScreenSpace;
+
+    public CombatModeIndicatorsOverlaySettings _settings = new(
+        Color.White.WithAlpha(0.3f),
+        Color.Black.WithAlpha(0.5f),
+        // 1 is a little big
+        0.8f
+    );
+
+    private Texture _gunSight;
+    private Texture _meleeSight;
+    private bool _isShowIndicators;
+
+    public ShowCombatModeIndicatorsOverlay(IConfigurationManager cfg, IInputManager input, IEntityManager entMan,
+            IEyeManager eye, CombatModeSystem combatSys)
+    {
+        IoCManager.InjectDependencies(this);
+
+        _cfg = cfg;
+        _inputManager = input;
+        _entMan = entMan;
+        _eye = eye;
+        _combatSystem = combatSys;
+
+        var spriteSys = _entMan.EntitySysManager.GetEntitySystem<SpriteSystem>();
+        _gunSight = spriteSys.Frame0(new SpriteSpecifier.Rsi(new($"/Textures/Interface/Misc/crosshair_pointers.rsi"),
+            "gun_sight"));
+        _meleeSight = spriteSys.Frame0(new SpriteSpecifier.Rsi(new($"/Textures/Interface/Misc/crosshair_pointers.rsi"),
+             "melee_sight"));
+
+        _isShowIndicators = _cfg.GetCVar(CCVars.HudHeldItemShow);
+
+        _cfg.OnValueChanged(CCVars.HudHeldItemShow,
+            isShow => _isShowIndicators = isShow, true);
+    }
+
+    protected override void Draw(in OverlayDrawArgs args)
+    {
+        if (!_isShowIndicators)
+            return;
+
+        if (!_combatSystem.IsInCombatMode())
+            return;
+
+        var mouseScreen = _inputManager.MouseScreenPosition;
+        var mousePosMap = _eye.ScreenToMap(mouseScreen);
+
+        if (mousePosMap.MapId == MapId.Nullspace)
+            return;
+
+        EntityUid? handEntity = _entMan.System<HandsSystem>().GetActiveHandEntity();
+        bool isHandGunItem = _entMan.HasComponent<GunComponent>(handEntity);
+
+        var screen = args.ScreenHandle;
+        var mousePos = mouseScreen.Position;
+        var uiScale = (args.ViewportControl as Control)?.UIScale ?? 1f;
+        float limitedScale = uiScale > 1.25f ? 1.25f : uiScale;
+
+        var sight = isHandGunItem ? _gunSight : _meleeSight;
+        DrawSight(sight, screen, mousePos, limitedScale * _settings.Scale);
+    }
+    private void DrawSight(Texture sight, DrawingHandleScreen screen, Vector2 centerPos, float scale)
+    {
+        screen.DrawTextureRect(sight,
+            GetRectForTexture(sight.Size, centerPos, scale), _settings.StrokeColor);
+        screen.DrawTextureRect(sight,
+            GetRectForTexture(sight.Size, centerPos, scale, true), _settings.MainColor);
+    }
+
+    static private UIBox2 GetRectForTexture(Vector2 textureSize, Vector2 pos, float scale,
+        bool expanded = false)
+    {
+        Vector2 sightSize = textureSize * scale;
+
+        if (expanded)
+            sightSize += 7;
+
+        return UIBox2.FromDimensions(pos - (sightSize / 2), sightSize);
+    }
```

Comment:
Just do this inline in the function.

Reply:
it will be too much for an inline function, so it looks better, in my opinion